lib/validate.js

```diff
@@ -14,8 +15,11 @@ var build = require('./build');
  */
 module.exports = (templatePath, region) => {
   var cfn = new AWS.CloudFormation({ region: region || 'us-east-1' });
-
-  return build(templatePath).then(function(template) {
+  return (
+    templatePath === '-' ? 
+      getStdin().then(JSON.parse) :
+      build(templatePath)
```

Comment:
One little point of confusion here -- it looks like if `templatePath` is given, then the expectation is that the data will be run through `build`, i.e. it could be .js. However the stdin approach wouldn't accept the same data -- it expects to be able to `JSON.parse` whatever is provided. 

Could the stdin flow allow for data piped from a `.js` file?

Reply:
> Could the stdin flow allow for data piped from a .js file?

Oooo... that would be tricky. If stdin were the contents of a .js file, we would need to either `eval` it or something (and so it in a way that matches what the module wrapper does) or write the contents to a temp file and `require` that.

The latter is feasible.

Yeah, we can do that. I think the stdin flow would then be:

1. read stdin into a string
2. try `JSON.parse` the string
3. if `JSON.parse` fails, write the string to a tempfile (with `.js` suffix) and then use that tempfile as `templatePath` passed to `build()`

Alternatively, we could _always_ create a tempfile with the contents of stdin, which would probably simplify the code at the cost of unnecessary I/O and file creation.

What do you think @rclark?